dvc/repo/fetch.py

```diff
@@ -22,7 +36,33 @@ def fetch(
             remote=remote,
             jobs=jobs,
             recursive=recursive,
-        )["local"]
-        return self.cloud.pull(
-            used, jobs, remote=remote, show_checksums=show_checksums
         )
+
+        downloaded = 0
+        failed = 0
+
+        try:
+            downloaded += self.cloud.pull(
+                used["local"],
+                jobs,
+                remote=remote,
+                show_checksums=show_checksums,
+            )
+        except NoRemoteError:
+            if not used["repo"]:
+                raise
+
+        except DownloadError as exc:
+            failed += exc.amount
+
+        for dep in used["repo"]:
+            try:
+                out = dep.fetch()
+                downloaded += out.get_files_number()
+            except DownloadError as exc:
```

Comment:
> Wrap dependency.repo.fetch to always return DownloadError (what about a git clone error? what about a find_out_by_relpath error? This will continue halting the repo.fetch operation)

For the record, for now we could simply add something ugly like:
```
except (CloneError, OutputNotFoundError) as exc:
    saved_exc = saved_exc or exc
...

if saved_exc:
    raise saved_exc
```
here to make it work. But there is probably a better solution, but it might require too much refactoring to make this look nice :slightly_frowning_face:

Reply:
@efiop , should we open an issue to refactor this one?

Reply:
@mroutis Sure.

Reply:
@efiop  https://github.com/iterative/dvc/issues/2522